src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetSourceRepositoryService.cs

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio.RemoteTypes;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class NuGetSourceRepositoryService : INuGetSourceRepositoryService
+    {
+        private bool _initialized = false;
+        private readonly AsyncLazyInitializer _initializer;
+        private ISourceRepositoryProvider _sourceRepositoryProvider;
+        private bool _disposedValue;
+
+        public event EventHandler PackageSourcesChanged;
```

Comment:
Don't think the event handler works correctly when remote. 
You'd probably need to define a new interface for that.

Reply:
Events on RPC interfaces *do* in fact work. But we have guidance for how to do them that you should be familiar with: https://dev.azure.com/devdiv/DevDiv/_git/VS.RPC.Contracts?path=%2Fdoc%2Fbestpractices.md&_a=preview